lib/driver.rb

```diff
@@ -43,5 +42,43 @@ def add_trip(trip)
 
       @trips << trip
     end
+
+    def finished_trips
+      trips_closed = @trips
+      trips_closed.each do |trip|
+        if trip.end_time == nil || trip.cost == nil || trip.rating == nil
```

Comment:
I love that you've broken this out as a separate helper method! This makes the other code in this file much easier to read.

Reply:
You could shorten this up a bit using an enumerable:

```ruby
def finished_trips
  return @trips.reject do |trip|
    trip.end_time == nil || trip.cost == nil || trip.rating == nil
  end
end
```